File: DLL/TrinitySeal/TrinitySeal/bin/Debug/SealCheck.vb
Imports System
Imports System.Collections.Generic
Imports System.IO
Imports System.Linq
Imports System.Security.Cryptography
Imports System.Text
Imports System.Threading.Tasks
Imports System.Windows.Forms
Imports System.Diagnostics

Namespace Seal
    Class SealCheck
        Public Shared isValidDLL As Boolean = False

        Public Shared Sub HashChecks()
            If CalculateMD5("Newtonsoft.Json.dll") <> "4df6c8781e70c3a4912b5be796e6d337" OrElse CalculateMD5(GetType(TrinitySeal.Seal).Assembly.Location) <> "0ac32b36f97427f59bd8e179c2594d95" Then
                MessageBox.Show("Hash check failed. Exiting...", "TrinitySeal", MessageBoxButtons.OK, MessageBoxIcon.[Error])
                Process.GetCurrentProcess().Kill()
            Else
                isValidDLL = True
            End If
        End Sub

        Private Shared Function CalculateMD5(ByVal filename As String) As String
            Using md = MD5.Create()

                Using stream = File.OpenRead(filename)
                    Dim hash = md.ComputeHash(stream)
                    Return BitConverter.ToString(hash).Replace("-", "").ToLowerInvariant()
                End Using
            End Using
        End Function
    End Class
End Namespace

File: DLL/TrinitySeal/TrinitySeal/bin/Debug/SealCheck.cs
using System;
using System.Collections.Generic;
using System.IO;
using System.Linq;
using System.Security.Cryptography;
using System.Text;
using System.Threading.Tasks;
using System.Windows.Forms;
using System.Diagnostics;

namespace TrinitySeal
{
    class SealCheck
    {
        public static bool isValidDLL = false;
        public static void HashChecks()
        {
            if (CalculateMD5("Newtonsoft.Json.dll") != "4df6c8781e70c3a4912b5be796e6d337" || CalculateMD5(typeof(TrinitySeal.Seal).Assembly.Location) != "0ac32b36f97427f59bd8e179c2594d95")
            {
                MessageBox.Show("Hash check failed! Exiting...", "TrinitySeal", MessageBoxButtons.OK, MessageBoxIcon.Error);
                Process.GetCurrentProcess().Kill();
            }
            else
            {
                isValidDLL = true;
            }
        }

        private static string CalculateMD5(string filename)
        {
            using (var md5 = MD5.Create())
            {
                using (var stream = File.OpenRead(filename))
                {
                    var hash = md5.ComputeHash(stream);
                    return BitConverter.ToString(hash).Replace("-", "").ToLowerInvariant();
                }
            }
        }
    }
}